cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -798,34 +797,33 @@ def _add_efs_storage(self, id: str, shared_efs: SharedEfs):
             efs_resource.cfn_options.deletion_policy = efs_resource.cfn_options.update_replace_policy = deletion_policy
             efs_id = efs_resource.ref
 
-        checked_availability_zones = []
-
-        # Mount Targets for Compute Fleet
-        compute_subnet_ids = self.config.compute_subnet_ids
-        file_system_security_groups = [self._add_storage_security_group(id, shared_efs)]
+            # Create Mount Targets
+            checked_availability_zones = []
+
+            # Mount Targets for Compute Fleet
+            compute_subnet_ids = self.config.compute_subnet_ids
+            file_system_security_groups = [self._add_storage_security_group(id, shared_efs)]
+
+            for subnet_id in compute_subnet_ids:
+                self._add_efs_mount_target(
```

Comment:
Previously we were creating Mount Targets, so with this change we're avoiding creation of both security groups and mount targets, is this expected?

Do you mean we had dangling Mount Targets as well? If so let's mention them in the changelog.

Reply:
No we didn't create the mount targets. Only the security groups were dangling.